src/engine.rs

```diff
@@ -43,8 +50,9 @@ pub struct Reedline {
     cut_buffer: Box<dyn Clipboard>,
 
     // History
-    history: History,
-    history_search: Option<BasicSearch>, // This could be have more features in the future (fzf, configurable?)
+    history: Box<dyn History>,
+    // TODO change this shit
```

Comment:
Remove this

Reply:
Can you say a bit more here?

Reply:
Also let's keep it "family friendly" as they say

Reply:
This was an TODO for me, earlier I had a boolean here to control some flow, which I was not a fan of. I removed the boolean but forgot to remove the comment. The Boxed history is fine with me too. 

Will keep the language in check :)